global-hooks/calculate_resources_requests_test.go

```diff
@@ -78,123 +72,54 @@ status:
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.memoryMaster").Int()).To(Equal(int64(512 * 1024 * 1024)))
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.milliCpuEveryNode").Int()).To(Equal(int64(500)))
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.memoryEveryNode").Int()).To(Equal(int64(1 * 1024 * 1024 * 1024)))
+			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.milliCpuControlPlane").Int()).To(Equal(int64(2000)))
+			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.memoryControlPlane").Int()).To(Equal(int64(2 * 1024 * 1024 * 1024)))
 		})
-
 	})
 
-	Context("Correctly set, two master nodes, global.modules.resourcesRequests.masterNode not set", func() {
+	Context("Correctly set, two master nodes, but without set global modules resourcesRequests", func() {
 		BeforeEach(func() {
 			f.BindingContexts.Set(f.KubeStateSet(stateMasterNode + stateMasterNode2))
-			f.ValuesSet("global.modules.resourcesRequests.everyNode.cpu", "300m")
-			f.ValuesSet("global.modules.resourcesRequests.everyNode.memory", "512Mi")
 			f.RunHook()
 		})
 
 		It("Hook should run and set global internal values", func() {
 			Expect(f).To(ExecuteSuccessfully())
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.milliCpuControlPlane").Int()).To(Equal(int64(874)))
+			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.milliCpuControlPlane").Int()).To(Equal(int64(850)))
 			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.memoryControlPlane").Int()).To(Equal(int64(1792 * 1024 * 1024)))
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.milliCpuMaster").Int()).To(Equal(int64(874)))
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.memoryMaster").Int()).To(Equal(int64(1792 * 1024 * 1024)))
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.milliCpuEveryNode").Int()).To(Equal(int64(300)))
-			Expect(f.ValuesGet("global.internal.modules.resourcesRequests.memoryEveryNode").Int()).To(Equal(int64(512 * 1024 * 1024)))
 		})
 
 	})
 
-	Context("Correctly set, two master nodes, global.modules.resourcesRequests.masterNode set", func() {
+	Context("Correctly set, two master nodes, with set global modules resourcesRequests", func() {
 		BeforeEach(func() {
 			f.BindingContexts.Set(f.KubeStateSet(stateMasterNode + stateMasterNode2))
-			f.ValuesSet("global.modules.resourcesRequests.everyNode.cpu", "500m")
-			f.ValuesSet("global.modules.resourcesRequests.everyNode.memory", "1Gi")
-			f.ValuesSet("global.modules.resourcesRequests.masterNode.cpu", "1")
-			f.ValuesSet("global.modules.resourcesRequests.masterNode.memory", "1Gi")
+			f.ValuesSet("global.modules.resourcesRequests.controlPlane.cpu", "1000m")
+			f.ValuesSet("global.modules.resourcesRequests.controlPlane.memory", "1Gi")
 			f.RunHook()
 		})
 
 		It("Hook should run and set global internal values", func() {
```

Comment:
Test claims are unclear to me. Which values are expected? And why? Are they from global values of from nodes? What is this particular case, and how is it different from other tested cases? 

Also, I have a suggestion how to get rid of all this "hook boilerplate". The decision-making function requires only 2 numbers + N nodes (2×N numbers) as input, and it returns two numbers as output. The tests would much more clean, itf they tested a decision-making function, not YAMLs, kube state and hooks framework and value s setting. Are you willing to make tests more clean?

Reply:
made the tests more transparent